File: psets/01/mario-more/mario.rb
puts "What's the height of pyramid you want?"

height = 0
height = $stdin.gets.chomp.to_i while height.negative? || height.zero? || height > 8

height.times do |i|
  i += 1
  # Printing space before hash:
  print(' ' * (height - i))

  # Printing left hashes before gap:
  print('#' * (height - (height - i)))

  # Printing gap:
  print('  ')

  # Printing right hashes after gap:
  print('#' * (height - (height - i)))

  # Printing newline:
  puts
end

File: psets/01/mario-more/mario.c
#include <cs50.h>
#include <stdio.h>

int main(void)
{
    // If the user doesn’t, in fact, input a positive integer between 1 and 8,
    // inclusive, when prompted, the program should re-prompt the user until ///// they cooperate:
    int height;
    do
    {
        height = get_int("Height:");
    } while (height < 1 || height > 8);
    for (int i = 0; i <= height; i++)
    {
        // Printing space before hash:
        for (int j = 0; j < (height - i); j++)
        {
            printf(" ");
        }
        // Printing left hashes before gap:
        for (int k = 0; k < i; k++)
        {
            printf("#");
        }
        // Printing gap:
        printf("  ");
        // Printing right hashes after gap:
        for (int l = 0; l < i; l++)
        {
            printf("#");
        }
        // Newline
        printf("\n");
    }
}
